homeassistant/setup.py

```diff
@@ -745,7 +757,26 @@ def async_start_setup(
     finally:
         time_taken = time.monotonic() - started
         del setup_started[current]
-        _setup_times(hass)[integration][group][phase] = time_taken
+        group_setup_times = _setup_times(hass)[integration][group]
+        # We may see the phase multiple times if there are multiple
+        # platforms, but we only care about the longest time.
+        group_setup_times[phase] = max(group_setup_times[phase], time_taken)
+        if group is None:
+            _LOGGER.info(
+                "Setup of domain %s took %.2f seconds", integration, time_taken
+            )
```

Comment:
Note there is no `return` here to save indent because we are in a finally and do not want to swallow any exceptions